recipes/dns.rb

```diff
@@ -1,5 +1,8 @@
 include_recipe 'dnsmasq::default'
-include_recipe 'dnsmasq::manage_hostsfile'
+
+if(node[:dnsmasq][:enable_hostsfile])
+  include_recipe 'dnsmasq::manage_hostsfile'
+end
```

Comment:
This proposal is stylistically inconsistent with existing conditional includes in `default.rb`; I preserved existing style, as consistency is more important for readability.